microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/SubsystemDefinition.java

```diff
@@ -79,4 +114,39 @@ public void registerAdditionalRuntimePackages(final ManagementResourceRegistrati
             resourceRegistration.registerAdditionalRuntimePackages(RuntimePackageDependency.required(m));
         }
     }
+
+    private static final class DefaultTracerWriteAttributeHandler extends ReloadRequiredWriteAttributeHandler {
+
+        private DefaultTracerWriteAttributeHandler() {
+            super(DEFAULT_TRACER);
+        }
+
+        @Override
+        protected void recordCapabilitiesAndRequirements(OperationContext context, AttributeDefinition attributeDefinition,
+                ModelNode newValue, ModelNode oldValue) {
+            super.recordCapabilitiesAndRequirements(context, attributeDefinition, newValue, oldValue);
```

Comment:
I don't think you want to call the super implementation, as you are handling capability registration conditionally.